bundles/plugins/org.bonitasoft.studio.common.repository/src/org/bonitasoft/studio/common/repository/core/ProjectManifestFactory.java

```diff
@@ -0,0 +1,86 @@
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.bonitasoft.engine.bpm.bar.BusinessArchive;
+import org.bonitasoft.studio.common.repository.CommonRepositoryPlugin;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.pde.internal.core.util.ManifestUtils;
+import org.osgi.framework.BundleReference;
+
+public class ProjectManifestFactory {
+
+    private static final String MANIFEST_MF_FILE_NAME = "MANIFEST.MF";
+    private static final String META_INF_FOLDER_NAME = "META-INF";
+
+    public void createProjectManifest(final IProject project, final IProgressMonitor monitor) throws CoreException {
+        final IFolder metaInf = project.getFolder(META_INF_FOLDER_NAME);
+        if (!metaInf.exists()) {
+            metaInf.create(true, true, monitor);
+        }
+        final IFile projectManifest = project.getFolder(META_INF_FOLDER_NAME).getFile(MANIFEST_MF_FILE_NAME);
+        if (!projectManifest.exists()) {
+            writeNewManifestFile(project, monitor, projectManifest);
+        }
+    }
+
+    protected void writeNewManifestFile(final IProject project, final IProgressMonitor monitor, final IFile projectManifest) throws CoreException {
+        try (PrintWriter writer = new PrintWriter(projectManifest.getLocation().toFile())) {
+            ManifestUtils.writeManifest(createManifestHeaders(project.getName()), writer);
+            projectManifest.refreshLocal(IResource.DEPTH_ONE, monitor);
+        } catch (final IOException e) {
+            throw new CoreException(new Status(IStatus.ERROR, CommonRepositoryPlugin.PLUGIN_ID, "Failed to create MANIFEST.MF", e));
+        }
+    }
+
+    protected Map<String, String> createManifestHeaders(final String projectName) {
+        final Map<String, String> headers = new HashMap<String, String>();
+        headers.put(ManifestUtils.MANIFEST_VERSION, "1.0");
+        headers.put(org.osgi.framework.Constants.BUNDLE_MANIFESTVERSION, "2");
+        headers.put(org.osgi.framework.Constants.BUNDLE_NAME, projectName);
+        headers.put(org.osgi.framework.Constants.BUNDLE_SYMBOLICNAME, toJavaIdentifier(projectName, false));
+        headers.put(org.osgi.framework.Constants.BUNDLE_VERSION, "1.0.0.qualifier");
+        headers.put(org.osgi.framework.Constants.BUNDLE_VENDOR, "BonitaSoft S.A.");
+        headers.put(org.osgi.framework.Constants.BUNDLE_REQUIREDEXECUTIONENVIRONMENT, "JavaSE-1.7");
+        headers.put(org.osgi.framework.Constants.REQUIRE_BUNDLE,
+                new StringBuilder()
+                        .append("javax.persistence;bundle-version=\"2.0.3\"").append(",")
```

Comment:
I don't if's a good idea (perf will be impacted)
and osgi version are not strict so if we update to a newer version it should not be impacted